docs/cpp-sdk-factory-design.md

```diff
@@ -94,6 +94,45 @@ This property makes it possible to:
 - deploy a new SDK shared library
 - keep the application unchanged
 
+### Case study, using Factory and shared gRPC client between OTLP gRPC exporters
+
+To reduce the cost of gRPC, the SDK allow users to shared gRPC clients between
+OTLP gRPC exporters when these exporters have the same settings. This can be
+used as follows from the application code:
+
+```cpp
+// Include follows headers
+#include "opentelemetry/exporters/otlp/otlp_grpc_client_factory.h"
+#include "opentelemetry/exporters/otlp/otlp_grpc_exporter_factory.h"
+#include "opentelemetry/exporters/otlp/otlp_grpc_exporter_options.h"
+#include "opentelemetry/exporters/otlp/otlp_grpc_log_record_exporter_factory.h"
+#include "opentelemetry/exporters/otlp/otlp_grpc_log_record_exporter_options.h"
+
+// Create exporters with shared gRPC Client
+namespace otlp = opentelemetry::exporter::otlp;
+
+void SetupOtlp() {
+  otlp::OtlpGrpcExporterOptions trace_opts;
+  otlp::OtlpGrpcLogRecordExporterOptions log_opts;
+
+  // Setting trace_opts and log_opts
+  std::shared_ptr<otlp::OtlpGrpcClient> shared_client =
+    otlp::OtlpGrpcClientFactory::Create(trace_opts);
+
+  // Create exporters
+  auto trace_exporter =
+    otlp::OtlpGrpcExporterFactory::Create(trace_opts, shared_client);
```

Comment:
Seems `shared_client` has been created with `trace_opts`, could the second `trace_opts` be ignored here and use the default on in `shared_client"?

Reply:
Future options, such as those for traces or logs, may not belong to the gRPC client, and the OTLP metrics exporter already includes a metrics-specific option. These options are still needed to create exporters, and having consistent creation parameters for all exporters makes them easier to understand.